Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -1438,6 +1450,11 @@ double FitPeaks::fitIndividualPeak(size_t wi, const API::IAlgorithm_sptr &fitter
   if (numberCounts(m_inputMatrixWS->histogram(wi), fitwindow.first, fitwindow.second) <= m_minPeakHeight)
     return cost;
 
+  // check if we have enough data points to fit
+  const int cushion = 2; // using a "cushion" here so that we have a bit more data points than parameters
```

Comment:
I forget the concept, but the "modern" way to do this is
```suggestion
  const std::size_t cushion{2}; // using a "cushion" here so that we have a bit more data points than parameters
```
The compiler will optimize it better, but I'm not sure it would make any difference for runtime